set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java

```diff
@@ -152,34 +145,31 @@
 	@ApiOperation(notes = SwaggerConstants.SEARCH_HELP_NOTE, value = "Retrieve a user set", 
 				nickname = "retrieve", response = java.lang.Void.class)
 	public ResponseEntity<String> getUserSet(
-			@RequestParam(value = WebUserSetFields.PARAM_WSKEY, required = false) String wskey,
 			@PathVariable(value = WebUserSetFields.PATH_PARAM_SET_ID) String identifier,
-			@RequestParam(value = WebUserSetFields.USER_TOKEN, required = false, defaultValue = WebUserSetFields.USER_ANONYMOUNS) String userToken,			
 			@RequestParam(value = WebUserSetFields.PROFILE, required = false, defaultValue = WebUserSetFields.PROFILE_MINIMAL) String profile,			
 			HttpServletRequest request) throws HttpException {
 
 		String action = "get:/set/{identifier}{.jsonld}";
-		return getUserSet(wskey, profile, identifier, request, action);
+		return getUserSet(profile, identifier, request, action);
 	}
 
 	/**
 	 * This method retrieves an existing user set identified by given identifier, which is
 	 * a number in string format.
-	 * @param wskey The API key
 	 * @param profile The profile definition
 	 * @param identifier The identifier
 	 * @param request HTTP request
 	 * @param action The action describing the request
 	 * @return response entity that comprises response body, headers and status code
 	 * @throws HttpException
 	 */
-	private ResponseEntity<String> getUserSet(String wsKey, String profileStr, String identifier, 
+	private ResponseEntity<String> getUserSet(String profileStr, String identifier, 
 			HttpServletRequest request, String action)
 					throws HttpException {
 		try {
 			// check user credentials, if invalid respond with HTTP 401.
-			// check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
+		    // verify access rights
+		    verifyReadAccess(request);
```

Comment:
this method must be invoked in the calling method

Reply:
verifyReadAccess invoked as suggested